### ISSUE TITLE
Crash when listener callback tries to remove itself

### ISSUE THREAD

**Issue by obcat** [vim/vim#8863]
**Describe the bug**

When a listener callback tries to remove itself, Vim crashes.

**To Reproduce**

1. Run `vim -Nu NONE`.

1. `:source` this script:

    ```vim
    let s:ID = listener_add('Listener')
    function Listener(...)
      call listener_remove(s:ID)
    endfunction
    put = 'text'
    ```

    Vim crashes.

**Expected behavior**

Vim does not crash. Not sure if the callback should be successfully removed or an error should be given.

**Environment**

- Vim 8.2.3424
- macOS 10.15.7
- iTerm2

**Comment by brammool** [comment]
Fixed by Patch 8.2.3426

**Comment by obcat** [comment]
Thank you.
